lib/models/patch/builder.js

```diff
@@ -1,31 +1,43 @@
-import {TextBuffer} from 'atom';
-
+import PatchBuffer from './patch-buffer';
 import Hunk from './hunk';
 import File, {nullFile} from './file';
-import Patch from './patch';
+import Patch, {TOO_LARGE, EXPANDED} from './patch';
 import {Unchanged, Addition, Deletion, NoNewline} from './region';
 import FilePatch from './file-patch';
 import MultiFilePatch from './multi-file-patch';
 
-export function buildFilePatch(diffs) {
-  const layeredBuffer = initializeBuffer();
+export const DEFAULT_OPTIONS = {
+  // Number of lines after which we consider the diff "large"
+  // TODO: Set this based on performance measurements
+  largeDiffThreshold: 100,
```

Comment:
@simurai: how many lines do you think constitutes a large diff?  Not just from a performance perspective, but from a user experience perspective.  Like how many lines is disruptive to a user when they're trying to read, because often large diffs are the result of auto generated code.

Reply:
Hmmm.. will large diffs be collapsed by default or there is a "load" button? Maybe if the diff is so large that it fills the whole scroll height. Then I can uncollapse only if I'm really interested in that file. 100 seems fine. 👍

Reply:
@vanessayuenn @kuychaco @smashwilson care to weigh in?

Reply:
mm we were using 1000 lines as the threshold before, but that's for performance reasons. 100 does seem a bit small though considering it's counting both deleted and added lines. 🤔 300 seems a bit more reasonable, but still an arbitrary number.

Reply:
> but still an arbitrary number.

Would this be good as a package option then? Pick a value that should work for most users, but let people who need to configure it. Also, would that be something that can be refined by `metrics`?